llm_bioinformatics_research/server.js

```diff
@@ -2,60 +2,133 @@ require('dotenv').config();
-  try {
-    await client.connect();
-    console.log("Connected to MongoDB");
-  } catch (error) {
-    console.error("Error connecting to MongoDB", error);
-  }
+    try {
+        await client.connect();
+        console.log("Connected to MongoDB");
+    } catch (error) {
+        console.error("Error connecting to MongoDB", error);
+    }
 }
 connectDB();
 
 app.post('/signup', async (req, res) => {
-  try {
-    const { user_name, email, password } = req.body; 
+    try {
+        const { user_name, email, password } = req.body;
+        const database = client.db("user_information");
+        const collection = database.collection("user_credentials");
 
-    const database = client.db("user_information");
-    const collection = database.collection("user_credentials");
+        const existingEmail = await collection.findOne({ email: email });
+        if (existingEmail) {
+            return res.status(400).json({ message: "Email is already in use" });
+        }
 
-    const existingEmail = await collection.findOne({ email: email });
-    if (existingEmail) {
-      return res.status(400).json({ message: "Email is already in use" });
-    }
+        const existingUserName = await collection.findOne({ user_name: user_name });
+        if (existingUserName) {
+            return res.status(400).json({ message: "Username is already in use" });
+        }
+
+        const hashedPassword = await bcrypt.hash(password, saltRounds);
 
-    const existingUserName = await collection.findOne({ user_name: user_name });
-    if (existingUserName) {
-      return res.status(400).json({ message: "Username is already in use" });
+        const result = await collection.insertOne({
+            user_name,
+            email,
+            password: hashedPassword
+        });
+
+        res.status(201).json(result);
+    } catch (error) {
+        res.status(500).json({ error: "Failed to insert user" });
     }
+});
 
-    const hashedPassword = await bcrypt.hash(password, saltRounds);
+app.post('/login', async (req, res) => {
+    const { identifier, password } = req.body;
 
-    const result = await collection.insertOne({
-      user_name, 
-      email, 
-      password: hashedPassword  
-    });
+    try {
+        const database = client.db("user_information");
+        const collection = database.collection("user_credentials");
 
-    res.status(201).json(result);  
+        const user = await collection.findOne({
+            $or: [{ email: identifier }, { user_name: identifier }]
+        });
 
-  } catch (error) {
-    res.status(500).json({ error: "Failed to insert user" });
-  }
-});
+        if (!user) {
+            return res.status(400).json({ message: "Invalid email/username or password" });
+        }
+
+        const isPasswordValid = await bcrypt.compare(password, user.password);
+        if (!isPasswordValid) {
+            return res.status(400).json({ message: "Invalid email/username or password" });
+        }
```

Comment:
Implementation is successful! Login credentials are correctly validated and user is directed to the home screen.